### PR TITLE
Switch blinded token requests to use new mywallet endpoint

### PR DESCRIPTION
**Pull request by tmancey**
Resolves https://github.com/brave/brave-browser/issues/21516

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

See https://github.com/brave/brave-browser/issues/21516

### CODE REVIEW

File: chromium_src/net/tools/transport_security_state_generator/input_file_parsers.cc [hunk 1]
```diff
@@ -457,14 +457,15 @@ bool ParseJSON(base::StringPiece json,
       { "name": "ropsten-infura.brave.com", "mode": "force-https", "policy": "custom", "pins": "brave"},
 
       // Rewards
-      { "name": "ads-serve.brave.com", "mode": "force-https", "policy": "custom", "pins": "brave"},
```
Comment (tmancey) [inline chromium_src/net/tools/transport_security_state_generator/input_file_parsers.cc:460]: Removed pinning for old host

File: chromium_src/net/tools/transport_security_state_generator/input_file_parsers.cc [hunk 2]
```diff
@@ -457,14 +457,15 @@ bool ParseJSON(base::StringPiece json,
...
       { "name": "ads-static.brave.com", "mode": "force-https", "policy": "custom", "pins": "brave"},
+      { "name": "anonymous.ads.brave.com", "mode": "force-https", "policy": "custom", "pins": "brave"},
       { "name": "api.rewards.brave.com", "policy": "custom", "mode": "force-https", "pins": "brave"},
       { "name": "api.rewards.bravesoftware.com", "policy": "custom", "mode": "force-https", "pins": "brave"},
       { "name": "api.rewards.brave.software", "policy": "custom", "mode": "force-https", "pins": "brave"},
       { "name": "grant.rewards.brave.com", "policy": "custom", "mode": "force-https", "pins": "brave"},
       { "name": "grant.rewards.bravesoftware.com", "policy": "custom", "mode": "force-https", "pins": "brave"},
       { "name": "grant.rewards.brave.software", "policy": "custom", "mode": "force-https", "pins": "brave"},
+      { "name": "mywallet.ads.brave.com", "mode": "force-https", "policy": "custom", "pins": "brave"},
```
Comment (tmancey) [inline chromium_src/net/tools/transport_security_state_generator/input_file_parsers.cc:468]: Added pinning for new non-anonymous host

File: chromium_src/net/tools/transport_security_state_generator/input_file_parsers.cc (hunk 1 above)
Comment (diracdeltas) [reply chromium_src/net/tools/transport_security_state_generator/input_file_parsers.cc:460]: are we not using this for any other hosts?

Comment (tmancey) [reply chromium_src/net/tools/transport_security_state_generator/input_file_parsers.cc:460]: This host is no longer used, thanks

File: vendor/bat-native-ads/src/bat/ads/internal/server/anonymous_server_host.cc [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+#include "bat/ads/internal/server/anonymous_server_host.h"
+
+#include "bat/ads/ads.h"
+#include "bat/ads/public/interfaces/ads.mojom.h"
+
+namespace ads {
+
+namespace {
+
+const char kStaticProductionHost[] = "https://ads-static.brave.com";
```
Comment (aseren) [inline vendor/bat-native-ads/src/bat/ads/internal/server/anonymous_server_host.cc:15]: nit: constexpr

File: vendor/bat-native-ads/src/bat/ads/internal/server/non_anonymous_server_host.cc [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+#include "bat/ads/internal/server/non_anonymous_server_host.h"
+
+#include "bat/ads/ads.h"
+#include "bat/ads/public/interfaces/ads.mojom.h"
+
+namespace ads {
+
+namespace {
+
+const char kStaticProductionHost[] = "https://ads-static.brave.com";
```
Comment (aseren) [inline vendor/bat-native-ads/src/bat/ads/internal/server/non_anonymous_server_host.cc:15]: nit: constexpr

File: vendor/bat-native-ads/src/bat/ads/internal/server/anonymous_server_host.cc (hunk 3 above)
Comment (tmancey) [reply vendor/bat-native-ads/src/bat/ads/internal/server/anonymous_server_host.cc]: Fixed :-)

File: vendor/bat-native-ads/src/bat/ads/internal/server/non_anonymous_server_host.cc (hunk 4 above)
Comment (tmancey) [reply vendor/bat-native-ads/src/bat/ads/internal/server/non_anonymous_server_host.cc]: Fixed :-)

**Comment by aseren** [comment]
LGTM

**Review by aseren — APPROVED** [review]

**Review by fmarier — APPROVED** [review]
Looks good to me.

File: vendor/bat-native-ads/src/bat/ads/internal/server/server_host_util.h [hunk 5]
```diff
@@ -1,21 +1,22 @@
...
+#define BRAVE_VENDOR_BAT_NATIVE_ADS_SRC_BAT_ADS_INTERNAL_SERVER_SERVER_HOST_UTIL_H_
 
 #include <string>
 
 namespace ads {
-namespace confirmations {
 namespace server {
 
-std::string GetHost();
+std::string GetStaticHost();
```
Comment (fmarier) [inline vendor/bat-native-ads/src/bat/ads/internal/server/server_host_util.h:14]: I'd suggest adding a comment above each of these to explain what they're for:

1. `GetStaticHost()`: Requests that are not user-specific and do not process personal data.
2. `GetGeoHost()`: Only used by the `/v1/getstate` endpoint.
3. `GetNonAnonymousHost()`: Requests that include the wallet ID and therefore fully identify the user.
4. `GetAnonymousHost()`: Requests that must not include the wallet ID or allow Brave to link the request to a wallet in any other way.

**Review by fmarier — APPROVED** [review]

**Review by fmarier — CHANGES_REQUESTED** [review]
Actually, could you also add `static.ads.brave.com` and `geo.ads.brave.com` to the list of pinned certificates?

File: vendor/bat-native-ads/src/bat/ads/internal/server/server_host_util.h (hunk 5 above)
Comment (tmancey) [reply vendor/bat-native-ads/src/bat/ads/internal/server/server_host_util.h:14]: Added

**Comment by tmancey** [comment]
@fmarier added missing hosts to pinned certificates

**Comment by tmancey** [comment]
@fmarier changed geo.ads.*.com to geolocation.ads.*.com as discussed

**Review by fmarier — APPROVED** [review]

**Pull request merged**